### ISSUE TITLE
JoSS Review: Functionality

### ISSUE THREAD

**Issue by betteridiot** [czbiohub/cerebra#106]
With regard to the ongoing review for your [JoSS submission](https://github.com/openjournals/joss-reviews/issues/2432), here are my notes/questions about the functionality of the package:

1. The hyperlink to [USAGE.md(https://github.com/czbiohub/cerebra/blob/master/docs/USAGE.md) within your [README.md](https://github.com/czbiohub/cerebra/blob/master/README.md) is not pointing to the correct place.
2. This might clear up my 3rd question: What does the `report_coverage` argument for `find-peptide-variants` do and why does it take an integer? If the argument is expecting a boolean, you can tell it to do so as such. If it has a range of options, those should be listed in the `-h` prompt.
3. If I used `1` for `report_coverage`, I got screen full of output that looked like `<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>`. This looks like somewhere in the code there is a `print()` on an object that does not have a representative `__repr__` or `__str__` methods. Therefore, to anyone that is not knee-deep in the code as a developer may be, this output means nothing to them. Consider defining those methods or making the output more meaningful.
4. What kind of file format is the output of `find-peptide-variants`? If it is a JSON (which I think you describe in the readme [section](https://github.com/czbiohub/cerebra#what-makes-cerebra-different-from-traditional-vcf-parsers)), maybe adding the `.json` suffix to the file path that the user supplies would be helpful. However, my `outfile` produced below does not seem to follow traditional JSON format (I see no brackets).

Below is my command history as detailed in your [README.md](https://github.com/czbiohub/cerebra/blob/master/README.md):

```bash
$ cerebra -h
Usage: cerebra  <command>

  a tool for fast and accurate summarizing of variant calling format (VCF)
  files

Options:
  -h, --help  Show this message and exit.

Commands:
  count-variants         count total number of variants in each sample, and...
  find-peptide-variants  report peptide-level SNPs and indels in each...
  germline-filter        filter out common SNPs/indels between...
$ cerebra germline-filter -h
Usage: cerebra germline-filter [OPTIONS]

  filter out common SNPs/indels between control/germline samples and samples
  of interest

Options:
  --processes INTEGER
  --control_path TEXT       [required]
  --experimental_path TEXT  [required]
  --metadata TEXT           [required]
  --outdir TEXT             [required]
  -h, --help                Show this message and exit.
$ cerebra count-variants -h
Usage: cerebra count-variants [OPTIONS] FILES...

  count total number of variants in each sample, and report on a per-gene
  basis

Options:
  --processes INTEGER
  --cosmicdb TEXT      [required]
  --refgenome TEXT     [required]
  --outfile TEXT       [required]
  -h, --help           Show this message and exit.
$ cerebra find-peptide-variants -h
Usage: cerebra find-peptide-variants [OPTIONS] INPUT_FILES...

  report peptide-level SNPs and indels in each sample, and associated
  coverage

Options:
  --processes INTEGER        number of processes to use for computation
  --cosmicdb TEXT            optional path to cosmic db file (.tsv)
  --annotation TEXT          path to genome annotation (.gtf)  [required]
  --genomefa TEXT            path to full genome sequences (.fasta)
                             [required]

  --report_coverage INTEGER  do you want to report coverage information?
                             [required]

  --output_path TEXT         path to output file  [required]
  -h, --help                 Show this message and exit.
$ cerebra germline-filter --processes 2 --control_path ./germline/ --experimental_path ./experimental/ --metadata meta.csv --outdir ./gl_out/
Running germline filter...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 18.90it/s]
Done!
$ cat A1.vcf
##fileformat=VCFv4.2
#CHROM  POS ID  REF ALT QUAL    FILTER  INFO
chr1    1   .   A   T,AT    0   .   .
chr1    5   .   C   T,G 0   .   .
chr1    18  .   GG  TAC,G,GC    0   .   .
$ cerebra find-peptide-variants ./vcf/* --processes 2 --cosmicdb cosmic_min.tsv --annotation gencode_min.gtf --genomefa GRCh38_limited_chr7.fa --report_coverage 1 --output_path ./outfile
Beginning setup (this may take several minutes!)
Loading COSMIC database...
Loading genome annotation...
Loading genome sequences...
Building genome trees...
Setup complete.
Finding mutations...
  0%|                                                                                                                                                                                         | 0/5 [00:00<?, ?it/s]<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 19.93it/s]
Writing file...
Done!
$ cat outfile
,EGFR
A1,"['ENSP00000275493.2:p.(Leu858Arg),[2:0]']"
A2,"['ENSP00000275493.2:p.(Leu858Arg),[3:2:1]', 'ENSP00000275493.2:p.(Leu858Arg),[0:2]']"
A3,[]
A4,[]
A5,[]
```

**Comment by lincoln-harris** [comment]
#3 is a known issue (#99) having to do with the biocommons/hgvs package. 
ill take another look at this

**Comment by lincoln-harris** [comment]
RE this cryptic error message: 
```
<class 'hgvs.edit.AASub'> <class 'hgvs.edit.AARefAlt'>
```
your right, this was an error message meant for internal debugging. 

shouldnt be seeing it anymore

**Comment by lincoln-harris** [comment]
2 and 4 should be addressed by the latest [PR](#109). basically the `report-coverage` option for `find-peptide-variants` returns a string that gives you some idea of the reads support for a given variant call. for example the string: 
```
'ENSP00000395243.3:p.(Leu813delinsArgTrp),[2:0]'
```
tells you you have the ENSP00000395243.3 variant with an insertion of _ArgTrp_ in place of _Leu_ at position 813. At this loci we find 2 variant reads and 0 corresponding to the wildtype loci.

**Comment by betteridiot** [comment]
Since it is a boolean, maybe setting up `click` to use a [flag](https://click.palletsprojects.com/en/7.x/options/#boolean-flags) instead would be a more intuitive approach: 

> Boolean flags are options that can be enabled or disabled. This can be accomplished by defining two flags in one go separated > by a slash (/) for enabling or disabling the option. (If a slash is in an option string, Click automatically knows that it’s a boolean > flag and will pass `is_flag=True` implicitly.) Click always wants you to provide an enable and disable flag so that you can change > the default later.
